File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        for(int i = 0; i < nums.size(); ++i) {
            int min_index = i; 
            for(int j = i; j < nums.size(); ++j) {
                if(nums[j] < nums[min_index]) {
                    min_index = j; 
                }
            }

            int temp = nums[i]; 
            nums[i] = nums[min_index];  
            nums[min_index] = temp; 
        }
    }
};

File: Data Structures & Algorithms/spiral-matrix/submission-0.py
class Solution:
    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
        top = 0
        bottom = len(matrix)-1
        left = 0
        right = len(matrix[0])-1

        nums = []


        i = 0
        j = 0

        d = 1

        while len(nums) < (len(matrix) * len(matrix[0])): 
            nums.append(matrix[i][j])

            if d == 1: 
                if j == right: 
                    d = 2
                    i += 1
                    top += 1
                else: 
                    j += 1

            elif d == 2: 
                if i == bottom: 
                    d = 3
                    j -= 1
                    right -= 1
                else: 
                    i += 1

            elif d == 3:
                if j == left: 
                    d = 4
                    i -= 1
                    bottom -= 1
                else: 
                    j -= 1

            else: 
                if i == top: 
                    d = 1
                    j += 1
                    left += 1
                else: 
                    i -= 1
        
        return nums